src/lavatory/utils/artifactory.py

```diff
@@ -153,31 +157,72 @@ def retain(self, spec_project, depth=3, terms=None, count=None, weeks=None):
         Returns:
             purgable (list): Purgable.
         """
-        if [terms, count, weeks].count(None) != 2:
+        if [terms, weeks].count(None) != 1:
             raise ValueError("Must specify exactly one of terms, count, or weeks")
 
-        purgable = []
-
+        purgable_artifacts = []
         for project in self.filter(depth=depth):
             if spec_project and spec_project != project["name"]:
                 continue
 
             path = "{}/{}".format(project["path"], project["name"])
-            if count:
-                filtered = self.filter(depth=depth + 1, terms=[{"path": path}])
-
-                for artifact in filtered:
-                    purgable.append("{}/{}".format(artifact["path"], artifact["name"]))
-
             if weeks:
                 now = datetime.datetime.now()
                 before = now - datetime.timedelta(weeks=weeks)
                 created = before.strftime("%Y-%m-%dT%H:%M:%SZ")
-                filtered = self.filter(depth=depth + 1, terms=[{"path": path}, {"created": created}])
-                for artifact in filtered:
-                    purgable.append("{}/{}".format(artifact["path"], artifact["name"]))
 
+                purgable_artifacts.extend(
+                    self.filter(offset=count, depth=depth + 1, terms=[{
```

Comment:
count variable is undefined.

Reply:
I'll fix this. This whole `retain` function I think will be removed as we refactor things out